python/lsst/resources/http.py

```diff
@@ -373,18 +374,81 @@ def is_webdav_endpoint(self) -> bool:
     def exists(self) -> bool:
         """Check that a remote HTTP resource exists."""
         log.debug("Checking if resource exists: %s", self.geturl())
-        resp = self.session.head(self.geturl(), timeout=TIMEOUT, allow_redirects=True)
-        return resp.status_code == 200
+        if not self.is_webdav_endpoint:
+            # The remote is a plain HTTP server. Let's attempt a HEAD
+            # request, even if the behavior for such a request against a
+            # directory is not specified, so it depends on the server
+            # implementation.
+            resp = self.session.head(self.geturl(), timeout=TIMEOUT, allow_redirects=True)
+            return resp.status_code == requests.codes.ok  # 200
+
+        # The remote endpoint is a webDAV server: send a PROPFIND request
+        # requesting only the 'getlastmodified' property.
+        request_body = """
+        <?xml version="1.0" encoding="utf-8" ?>
+        <D:propfind xmlns:D="DAV:"><D:prop><D:getlastmodified/></D:prop></D:propfind>
+        """
+        resp = self._propfind(request_body)
+        if resp.status_code == requests.codes.multi_status:  # 207
+            # Parse the XML-encoded response. Since we issue a request for a
+            # single remote resource, the response must include properties for
+            # that single resource. So either its response status is OK or not.
+            tree = etree.fromstring(resp.content.strip())
+            element = tree.find("./{DAV:}response/{DAV:}propstat/[{DAV:}status='HTTP/1.1 200 OK']")
```

Comment:
A question for my edification: can this ever be a different HTTP version (e.g. 1.0 or 2.0)?

Reply:
My understanding of the [RFC 4918](http://www.webdav.org/specs/rfc4918.html#n-propfind-status-codes)is that the response to `PROPFIND` requires that `status` be of the form `HTTP/1.1 <status code> <message>`.

But I would say this does not preclude that the communication protocol used between client and server be HTTP2 or HTTP3.